Larger flash support for QT Py

veng1 opened the pull request:
Please add support for larger capacity flash chips on the QT Py.  It appears that up to 128M is supported in the 8 pin SOP.

jepler committed:
_canio: Minimal implementation for SAM E5x MCUs

Tested & working:

 * Send standard packets
 * Receive standard packets (1 FIFO, no filter)

Interoperation between SAM E54 Xplained running this tree and
MicroPython running on STM32F405 Feather with an external
transceiver was also tested.

Many other aspects of a full implementation are not yet present,
such as error detection and recovery.

jepler committed:
samd: move mcu_find_pin_function to header

.. it'll be used in can as well as sdio

jepler committed:
sdioio: fix code for the case where there is no SDHC1

.. it doesn't really make a difference (the old code created an empty
else{} statement) but this is more correct.

jepler committed:
samd: SPI: improve conditional code

I recently misdiagnosed a "maybe-uninitialized" diagnostic as a bug in
asf4.  However, the problem was in our SPI code.

A special case for samr21 MCUs was being applied to same54p20a and possibly
other D5x/E5x MCUs, since the check was simply for pin PC19 existing at all.

Change the check to use the macro PIN_PC19F_SERCOM4_PAD0 which is only
defined if special function F of pin PC19 is SERCOM4 PAD0.

Reorganize the code a little bit so that brace-matching in editors is
not confused by the conditionalized code, including an unrelated change
for APA102_SCK's condition.

Revert the change to the Makefile that incorrectly attempted to silence
the diagnostic.

jepler committed:
supervisor: Improve serial connection detection

These changes remove the caveat from supervisor.runtime.serial_connected.

It appears that _tud_cdc_connected() only tracks explicit changes to the
"DTR" bit, which leads to disconnects not being registered.

Instead:
 * when line state is changed explicitly, track the dtr value in
   _serial_connected
 * when the USB bus is suspended, set _serial_connected to False

Testing performed (using sam e54 xplained):  Run a program to show
the state of `serial_connected` on the LED:
```
import digitalio
import supervisor
import board

led = digitalio.DigitalInOut(board.LED)
while True:
    led.switch_to_output(not supervisor.runtime.serial_connected)
```

Try all the following:
 * open, close serial terminal program
    - LED status tracks whether terminal is open
 * turn on/off data lines using the switchable charge-only cable
    - LED turns off when switch is in "charger" position
    - LED turns back on when switch is in Data position and terminal is
      opened (but doesn't turn back on just because switch position is
      changed)

jepler committed:
sam e54: disable microcontroller.cpu.voltage

This hangs, and the usual workarounds didn't work.

DavePutz committed:
Update from adafruit/main

DavePutz committed:
Sync with adafruit/main

hierophect committed:
Fix toctree, incorrect flash port assumptions

tannewt committed:
Merge pull request #3397 from tannewt/fix_size_0_bug

Fix bug with socket.recv_into size == 0.

DavePutz committed:
Merge pull request #27 from adafruit/main

Update from adafruit/main

DavePutz committed:
Merge pull request #28 from adafruit/main

Merge from adafruit/main

tannewt committed:
Merge pull request #3392 from hierophect/esp32-displayio-fix

Fix some DisplayIO crashes, add ESP32-S2 debugging capability

jepler committed:
update submodules to merge commits

dhalbert committed:
Merge pull request #3384 from jepler/feather_m4_can

atmel-samd: add same51, feather_m4_can

askpatrickw committed:
RTC enabled

jepler committed:
supervisor translate: explain the dictionary

askpatrickw committed:
code review feedback

askpatrickw committed:
Revert "code review feedback"

This reverts commit f9f614b3a2878383cad845f72ffef034bd2be951.

tannewt committed:
Add more PIDs for unique boards

tannewt committed:
Add partition layouts for 8 and 16 MB as well.

hugodahl committed:
Translated using Weblate (French)

Currently translated at 94.7% (753 of 795 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/fr/

astrobokonon committed:
Add BSSID to Network

tawez committed:
Translated using Weblate (Polish)

Currently translated at 71.3% (567 of 795 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pl/

remisarrailh committed:
Add Muselab nanoesp32 s2 definition

remisarrailh committed:
Fix invalid indentation on build.yml

kamtom480 committed:
camera: Pass width and height to take_picture()

remisarrailh committed:
Fix name in build.yml

jepler committed:
makeqstrdata: comment my understanding of @ciscorn's code

remisarrailh committed:
Change build-arm to build-xtensa in build.yml

jepler committed:
Merge pull request #3237 from DavePutz/issue3216

Issue#3216 - Needed an alternate way to check if background tasks were run

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/

hierophect committed:
Clarify location of port root

wtuemura committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (796 of 796 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pt_BR/

dhalbert committed:
Merge pull request #3369 from kamtom480/camera

Add camera module

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/

jepler committed:
py: Add enum helper code

This makes it much easier to implement enums, and the printing code is
shared.  We might want to convert other enums to this in the future.

jepler committed:
_canio: Message: Setting data should set the size to match

jepler committed:
Listener: fix doc markup, include in globals of its module

jepler committed:
_canio: Add _canio.State enum-like class

jepler committed:
_canio.CAN: add error handling & bus state

jepler committed:
correct docstring

askpatrickw committed:
RTC is pickd up automatically

jepler committed:
more doc improvements

jepler committed:
main.c: Correct CANIO if-guard

hierophect committed:
Add SPI memset, optional flags

tannewt committed:
Merge pull request #3398 from jepler/better-dictionary-compression

compression: Implement @ciscorn's dictionary approach

tannewt committed:
Merge pull request #3422 from askpatrickw/rtc-support

RTC enabled

tannewt committed:
Merge pull request #3427 from usini/muselab_nanoesp32_s2

Added board - Muselab nanoESP32-S2

wtuemura committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (799 of 799 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pt_BR/

bd34n committed:
rename board bdmicro_vina_m0 to bdmicro_vina_d21 and update boardfiles appropriately

bd34n committed:
pins.c: fix trailing whitespace

astrobokonon committed:
connect now accepts bssid

astrobokonon committed:
Add new error msg to translation

astrobokonon committed:
Merge branch 'main' into esp32s2-bssid

tawez committed:
Translated using Weblate (Polish)

Currently translated at 73.2% (585 of 799 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pl/

jepler committed:
Message: add extended address flag

jepler committed:
_canio: Add listener matching

Lightly tested:
 * no matches (catch-all)
 * standard address single address matches (even and odd positions)
 * standard address mask matches
 * only tested that extended doesn't match non-extended

jepler committed:
fix whitespace

jepler committed:
Whitespace

jepler committed:
Respond to review comments

Thanks @tannewt!

tannewt committed:
Merge pull request #3417 from hierophect/esp32s2-readme

ESP32S2: Add Readme

tannewt committed:
Merge pull request #3393 from tannewt/fix_spi_psram

Add non-DMA SPI support.

tannewt committed:
Merge pull request #2966 from sommersoft/ci_check_vid_pid

Automate USB VID/PID Uniqueness Check

tannewt committed:
Merge pull request #3426 from tannewt/esp_flash_sizes

Add partition layouts for 8 and 16 MB as well.

astrobokonon committed:
Add else case for non-bssid usage

tannewt committed:
Reorganize sdkconfigs and release UF2s

nitz committed:
Add SDA & SDL, RX & TX to pins.c

Despite the [silk on the dock board](https://wiki.makerdiary.com/nrf52840-m2-devkit/resources/nrf52840_m2_devkit_hw_diagram_v1_0.pdf), the SDA/SCL pins weren't defined. Though, they were already defined in `mpconfigboard.h`.

Same for RX/TX. It looks like it declared `TXD` and `RXD`, so I didn't want to remove those, but I think it makes sense to have the "standard" pin names, but I moved ithem to illustrate they were all referencing the same pins.

I mimicked the whitespace I saw in the metro_nrf52840_express port.

nitz committed:
Removed TXD/RXD, fixed whitespace.

jepler committed:
Add CAN to SAM E51 clock config

jepler committed:
add _canio to Feather M4 CAN

jepler committed:
_canio: Message: setting data clears rtr, and vice versa

hierophect committed:
Add never_reset reservation to RGBMatrix init

hierophect committed:
Remove timer debug messages

dhalbert committed:
Merge pull request #3437 from jepler/serial-connected-fix

supervisor: Improve serial connection detection

nitz committed:
More pin names cleanup.

jepler committed:
samd: template.ld: put canram in its own output section

jepler committed:
canio: Make both tx and rx (but not both) optional

.. loopback and silent come from the (optional) constructor parameters
not guessing based on the pin specification

.. docstring & comment improvements

dhalbert committed:
Merge pull request #3434 from jepler/same54-disable-cpu-voltage

sam e54: disable microcontroller.cpu.voltage

nitz committed:
Update user button names.

dhalbert committed:
Merge pull request #3435 from jepler/fix-sdhc1

sdioio: fix code for the case where there is no SDHC1

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (799 of 799 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/es/

dhalbert committed:
Merge pull request #3436 from jepler/fix-spi-sercom4

samd: SPI: improve conditional code

dhalbert committed:
Merge pull request #3431 from hierophect/stm32-spi-writevalue

STM32: Change SPI Read to acknowledge write_value

dhalbert committed:
Merge pull request #3433 from astrobokonon/esp32s2-bssid

ESP32S2: Allow connecting to specific bssid

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/

dhalbert committed:
Merge pull request #3438 from nitz/patch-1

[Makerdiary nRF52840 M.2 Devkit] Add SDA & SDL, RX & TX to pins.c

wtuemura committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (800 of 800 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pt_BR/

jepler committed:
makeqstrdata: Work with older Python

This construct (which I added without sufficient testing,
apparently) is only supported in Python 3.7 and newer.  Make it
optional so that this script works on other Python versions.  This
means that if you have a system with non-UTF-8 encoding you will
need to use Python 3.7.

In particular, this affects a problem building circuitpython in
github's ubuntu-18.04 virtual environment when Python 3.7 is not
explicitly installed.  cookie-cuttered libraries call for Python
3.6:
```
    - name: Set up Python 3.6
      uses: actions/setup-python@v1
      with:
        python-version: 3.6
```
Since CircuitPython's own build calls for 3.8, this problem was not
detected.

This problem was also encountered by discord user mdroberts1243.

The failure I encountered was here:
https://github.com/jepler/Jepler_CircuitPython_udecimal/runs/1138045020?check_suite_focus=true
.. while my step of "clone and build circuitpython unix port" is
unusual, I think the same problem would have affected "build assets"
if that step had been reached.

dhalbert committed:
Merge pull request #3432 from bd34n/rename-bdmicro-vina-m0-d21

Rename board bdmicro_vina_m0 to bdmicro_vina_d21 and update boardfiles.

dhalbert committed:
Merge pull request #3428 from weblate/weblate-circuitpython-main

Translations update from Weblate

jepler committed:
Merge pull request #3441 from hierophect/stm32-rgbmatrix-crashfix

STM32: Add never_reset reservation to RGBMatrix init

dhalbert committed:
Merge pull request #3443 from jepler/python-version-compat

makeqstrdata: Work with older Python

microdev1 committed:
Update microS2 config files

jepler committed:
makeqstrdata: Work around python3.6 compatibility problem

Discord user Folknology encountered a problem building with Python 3.6.9,
`TypeError: ord() expected a character, but string of length 0 found`.

I was able to reproduce the problem using Python3.5*, and discovered that
the meaning of the regular expression `"|."` had changed in 3.7.  Before,
```
>>> [m.group(0) for m in re.finditer("|.", "hello")]
['', '', '', '', '', '']
```
After:
```
>>> [m.group(0) for m in re.finditer("|.", "hello")]
['', 'h', '', 'e', '', 'l', '', 'l', '', 'o', '']
```
Check if `words` is empty and if so use `"."` as the regular expression
instead.  This gives the same result on both versions:
```
['h', 'e', 'l', 'l', 'o']
```
and fixes the generation of the huffman dictionary.

Folknology verified that this fix worked for them.

 * I could easily install 3.5 but not 3.6.  3.5 reproduced the same problem

tannewt committed:
Merge pull request #3445 from microDev1/boardInfo

Update microS2 config files

tannewt committed:
Merge pull request #3446 from jepler/python35-compression-fix

makeqstrdata: Work around python3.5/3.6 compatibility problem

hierophect committed:
Use write_value, add missing pin exceptions

jepler committed:
remove unneeded symbols

jepler committed:
canio: Message: remove size property

This is just duplicating what len(message.data) gives you.

jepler committed:
common.template.ld: remove unneeded line

jepler committed:
canio.CAN: switch rx/tx, make both mandatory, move declarations around

jepler committed:
CAN.listen: block until a message comes in or timeout is reached

jepler committed:
canio: rename from _canio

This reflects our belief that the API is stable enough to avoid incompatible changes during 6.x.

jepler committed:
make translate

jepler committed:
Fix unintentional cross references

jepler committed:
canio: a few more stray _canios

mdroberts1243 committed:
Initial commit bool column_and_page_addressing

FoamyGuy committed:
Merge branch 'adafruit_main' into sdcard_odb_fix

mdroberts1243 committed:
Quirk coded up for ...set_region_to_update

mdroberts1243 committed:
Changes to compile cleanly

microdev1 committed:
Update port.c

szajakubiak committed:
Translated using Weblate (Polish)

Currently translated at 73.3% (587 of 800 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pl/

hierophect committed:
Merge pull request #3451 from microDev1/patch-1

FIX : esp32s2 RTC support

jepler committed:
canio: Fix implementation bugs in atmel-sam

jepler committed:
canio: fix bugs in shared-bindings

DavePutz committed:
Merge pull request #29 from adafruit/main

Update from adafruit/main

siehputz committed:
Changed delay calls to avoid conflicts with qspi_disable

jepler committed:
canio: fix read(), iterable bindings

mdroberts1243 committed:
Removed redundant send of page setting

UnexpectedMaker committed:
Updated UM boards

UnexpectedMaker committed:
Updated UM boards

tannewt committed:
Merge pull request #3425 from jepler/canbus

canio: Initial implementation for SAM E5x MCUs

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/

wtuemura committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (807 of 807 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pt_BR/

jepler committed:
Merge pull request #3452 from weblate/weblate-circuitpython-main

Translations update from Weblate

jepler committed:
makeqstrdefs: don't make _and_, _or_ poisoned substrings for QSTRs

New contributor @mdroberts1243 encountered an interesting problem in
which the argument they had named "column_underscore_and_page_addressing"
simply couldn't be used; I discovered that internally this had been
transformed into "column_underscore∧page_addressing", because QSTR
makes _ENTITY_ stand for the same thing as &ENTITY; does in HTML.

This might be nice for some things, but we don't want it here!
I was unable to find a sensible way to "escape" and prevent this entity
coding, so instead I ripped out support for the _and_ and _or_ escapes.

tawez committed:
Translated using Weblate (Polish)

Currently translated at 73.7% (595 of 807 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pl/

ladyada committed:
fix pins for rev C

ladyada committed:
neopix is default on!

hierophect committed:
Move missing pin warning to shared-bindings

tannewt committed:
Merge pull request #3457 from ladyada/qtpy_m0

fix pins for rev C

tannewt committed:
Merge pull request #3456 from jepler/qstr-and-or-demagic

makeqstrdefs: don't make _and_, _or_ poisoned substrings for QSTRs

tannewt committed:
Merge pull request #3448 from hierophect/esp32-sd-fix

ESP32S2 - Fix SPI's SD card issue, add pin protections

ciscorn committed:
Translated using Weblate (Japanese)

Currently translated at 73.2% (591 of 807 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/ja/

jepler committed:
canio: doc improvements

jepler committed:
Add some NORETURN attributes

I have a function where it should be impossible to reach the end, so I put in a safe-mode reset at the bottom:
```
int find_unused_slot(void) {
    // precondition: you already verified that a slot was available
    for (int i=0; i<NUM_SLOTS; i++) {
        if( slot_free(i)) {
            return i;
        }
    }
    safe_mode_reset(MICROPY_FATAL_ERROR);
}
```
However, the compiler still gave a diagnostic, because safe_mode_reset was not declared NORETURN.

So I started by teaching the compiler that reset_into_safe_mode never returned.  This leads at least one level deeper due to reset_cpu needing to be a NORETURN function.  Each port is a little different in this area.  I also marked reset_to_bootloader as NORETURN.
Additional notes:

 * stm32's reset_to_bootloader was not implemented, but now does a bare reset.  Most stm32s are not fitted with uf2 bootloaders anyway.
 * ditto cxd56
 * esp32s2 did not implement reset_cpu at all.  I used esp_restart().  (not tested)
 * litex did not implement reset_cpu at all.  I used reboot_ctrl_write.  But notably this is what reset_to_bootloader already did, so one or the other must be incorrect (not tested).  reboot_ctrl_write cannot be declared NORETURN, as it returns unless the special value 0xac is written), so a new unreachable forever-loop is added.
 * cxd56's reset is via a boardctl() call which can't generically be declared NORETURN, so a new unreacahble "for(;;)" forever-loop is added.
 * In several places, NVIC_SystemReset is redeclared with NORETURN applied.  This is accepted just fine by gcc.  I chose this as preferable to editing the multiple copies of CMSIS headers where it is normally declared.
 * the stub safe_mode reset simply aborts.  This is used in mpy-cross.

jepler committed:
Match: address -> id

jepler committed:
Listerner: read -> receive, drop readinto

This prepares for creating a separate RemoteTransmissionRequest class

jepler committed:
canio: Split RemoteTransmissionRequest to its own class

It reuses most of canio.Message's implementation, and structure

jepler committed:
make translate

jepler committed:
supervisor: stub: make unimplemented safe_mode loop forever

jepler committed:
qtpy_m0_haxpress: add

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (807 of 807 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/es/

mdroberts1243 committed:
Mostly-working-version with comments

mdroberts1243 committed:
Fixed page set mask to be four bits!

askpatrickw committed:
Print package if it exists for pin or module only if it does not

tannewt committed:
Change I2C default to 100khz

Greater that 100khz is technically out of the original spec.

Background here: https://github.com/adafruit/Adafruit_CircuitPython_CLUE/issues/36

tannewt committed:
Merge pull request #3455 from weblate/weblate-circuitpython-main

Translations update from Weblate

tannewt committed:
Merge pull request #3470 from askpatrickw/fix-pin-printout

Print package if it exists for pin or module only if it does not

jepler committed:
esp32: Use esp_restart from reset_to_bootloader; redeclare it NORETURN

jepler committed:
mimxrt10xx: Add required header for NORETURN definition

tannewt committed:
Update TinyUSB to get MIDI SysEx fix

Fixes #3465

tannewt committed:
Merge pull request #3471 from tannewt/i2c_100k

Change I2C default to 100khz

mdroberts1243 committed:
Requested changes take 1

mdroberts1243 committed:
Changes take 2: missed ugame10 board

jepler committed:
canio: Listener: shuffle function declarations around

jepler committed:
canio: RemoteTransmissionRequest: Split implementation, keep one structure

This already begins obscuring things, because now there are two sets of
shared-module functions for manipulating the same structure, e.g.,
common_hal_canio_remote_transmission_request_get_id and
common_hal_canio_message_get_id

jepler committed:
canio: docs: fix how we refer to 'the listen method'

jepler committed:
canio: Correct type annotations of CAN.send, Listener.receive

jepler committed:
qtpy haxpress: enable RTC

jepler committed:
add new board to actions build

jepler committed:
give qtpy haxpress its own USB PID

jepler committed:
Merge pull request #3476 from jepler/qtpy-haxpress

Add QT Py Haxpress

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/

jepler committed:
Merge pull request #3474 from tannewt/update_tinyusb

Update TinyUSB to get MIDI SysEx fix

ansonhex committed:
Default power on Wio Terminal's 40 Pin

caternuson committed:
add MP const info to Design Guide

caternuson committed:
RST tweak

jepler committed:
canio: CAN.listen: rename argument to 'matches', reflecting that it is a sequence

jepler committed:
canio: CAN.Match: improve how an unspecified `mask` is implemented

0 should actually indicate a "match everything" filter which you otherwise
couldn't indicate with any single Match object, and an
all-address-bits-set number should indicate a "match single address"
filter.  Use an optional/default None argument to do the job.

tannewt committed:
Merge pull request #3453 from DavePutz/issue3367

Issue #3367 Changed delay calls to avoid conflicts with qspi_disable

tannewt committed:
Merge pull request #3481 from caternuson/iss3480_docupdate

Add MP const info to Design Guide

tannewt committed:
Merge pull request #3478 from ansonhe97/main

Turn Default power on Wio Terminal's 40 Pin

tannewt committed:
One grep

jepler committed:
CAN: Use mp_obj_t insteaed of canio_message_obj_t, get rid of rtr field

jepler committed:
canio: remove stray remote_transmission_request method declaration

jepler committed:
make translate

tawez committed:
Translated using Weblate (Polish)

Currently translated at 73.8% (595 of 806 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pl/

hierophect committed:
Change submodule, rework all includes

mdroberts1243 committed:
modified _stage/__init__.c call to set region to update

mdroberts1243 committed:
Merge branch 'main' into New_quirk_for_SH1107

jepler committed:
rgbmatrix: validate width= constructor parameter

In #3482, @cwalther noted that, hypothetically, a zero byte allocation
could be made in the RGBMatrix constructor.  Ensure that width is positive.
Height was already checked against the number of RGB pins if it was specified,
so zero is ruled out there as well.

jepler committed:
rgbmatrix: Check that the number of rgb pins is supported.

Having zero RGB pins may not have been caught, nor having a non-multiple-of-6
value.  Generally, users will only have 6 RGB pins unless they are driving
multiple matrices in parallel.  No existing breakouts exist to do this, and
there are probably not any efficient pinouts to be had anyway.

jepler committed:
can: RemoteTransmissionRequest: remove duplicate docstrings

jepler committed:
more doc improvements

jepler committed:
Merge pull request #3475 from tannewt/esp_uf2

Reorganize sdkconfigs and release UF2s

microdev1 committed:
Update make translate script

jepler committed:
Allow the _state_count properties to be unimplemented

This is the case on the SPI implementations MCP2515 and MCP2518.

jepler committed:
canio: actually drop the _error_count properties

thanks @tannewt

jepler committed:
canio: doc improvement -- don't needlessly call out deinit behavior

jepler committed:
Merge pull request #3486 from microDev1/fixTranslate

Update make translate script

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/

mdroberts1243 committed:
Ran pre-commit locally

tannewt committed:
Add Metro ESP32S2

Also fix two bugs:
* Crash when resetting board.SPI when it hasn't been inited.
* Reading back the output value is always false. Fixes #3353

astrobokonon committed:
Make sure you're a STA before connecting

wtuemura committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (826 of 826 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pt_BR/

mdroberts1243 committed:
Removed redundant subrectangle sizing code

UnexpectedMaker committed:
Removed ProS2 board files for now

jepler committed:
Makefile: Correct and expand translation excluded directories

As originally written, the TRANSLATE_SOURCES_EXC was incorrect.  If
more than one build directory existed, "make translate"
(and make check-translate) would error.

I corrected the problem, commented it, and added a number of additional
exclude directives.

I reviewed the PR and should have caught this, but did not.

jepler committed:
canio.CAN: clarify read-only status of several properties

jepler committed:
Merge remote-tracking branch 'origin/main' into canbus-api-changes

bergdahl committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (826 of 826 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/sv/

tannewt committed:
Ignore duplicate USB PID for prerelease version

tannewt committed:
Fix trailing line

tannewt committed:
Merge pull request #3472 from weblate/weblate-circuitpython-main

Translations update from Weblate

tannewt committed:
Merge pull request #3466 from jepler/canbus-api-changes

Canbus: api changes

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/

FoamyGuy committed:
removing top level displayio background check

tannewt committed:
Merge pull request #3450 from mdroberts1243/New_quirk_for_SH1107

New quirk for sh1107

tannewt committed:
Merge pull request #3489 from astrobokonon/esp32s2-fixconnection

ESP32S2: Enter station mode before connecting

tannewt committed:
Add to board list

hierophect committed:
Revert HAL refactor changes to avoid USB breaking changes past esp-idf dde6222cd

UnexpectedMaker committed:
Merge pull request #1 from tannewt/UM_S2_Boards

Um s2 boards

FoamyGuy committed:
remove remaining top level displayio background check

wtuemura committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (827 of 827 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pt_BR/

jepler committed:
Merge pull request #3494 from weblate/weblate-circuitpython-main

Translations update from Weblate

jepler committed:
sharpdisplay: Fix memory corruption across soft-reset

It was incorrect to NULL out the pointer to our heap allocated buffer in
`reset`, because subsequent to framebuffer_reset, but while
the heap was still active, we could call `get_bufinfo` again,
leading to a fresh allocation on the heap that is about to be destroyed.

Typical stack trace:
```
#1  0x0006c368 in sharpdisplay_framebuffer_get_bufinfo
#2  0x0006ad6e in _refresh_display
#3  0x0006b168 in framebufferio_framebufferdisplay_background
#4  0x00069d22 in displayio_background
#5  0x00045496 in supervisor_background_tasks
#6  0x000446e8 in background_callback_run_all
#7  0x00045546 in supervisor_run_background_tasks_if_tick
#8  0x0005b042 in common_hal_neopixel_write
#9  0x00044c4c in clear_temp_status
#10 0x000497de in spi_flash_flush_keep_cache
#11 0x00049a66 in supervisor_external_flash_flush
#12 0x00044b22 in supervisor_flash_flush
#13 0x0004490e in filesystem_flush
#14 0x00043e18 in cleanup_after_vm
#15 0x0004414c in run_repl
#16 0x000441ce in main
```
When this happened -- which was inconsistent -- the display would keep
some heap allocation across reset which is exactly what we need to avoid.

NULLing the pointer in reconstruct follows what RGBMatrix does, and that
code is a bit more battle-tested anyway.

If I had a motivation for structuring the SharpMemory code differently,
I can no longer recall it.

Testing performed: Ran my complicated calculator program over multiple
iterations without observing signs of heap corruption.

Closes: #3473

Commit:
Translated using Weblate (Spanish)

Currently translated at 99.8% (826 of 827 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/es/

jepler committed:
Update shared-bindings/rgbmatrix/RGBMatrix.c

Co-authored-by: Scott Shawcroft <scott@tannewt.org>

jepler committed:
Makefile: translate: exclude e.g., ports/unix/build as well

tannewt committed:
Merge pull request #3496 from weblate/weblate-circuitpython-main

Translations update from Weblate

tannewt committed:
Merge pull request #3449 from FoamyGuy/sdcard_odb_fix

Sdcard odb fix

hierophect committed:
Update TinyUSB include compatibility

tannewt committed:
Merge pull request #3495 from tannewt/esp32s2_metro

Add Metro ESP32S2

tannewt committed:
Merge pull request #3463 from UnexpectedCircuitPython/UM_S2_Boards

Updated UM FeatherS2 boards

tannewt committed:
Merge pull request #3497 from jepler/issue3473

sharpdisplay: Fix memory corruption across soft-reset

tannewt committed:
Merge branch 'main' into framebuffer-zero-size

tannewt committed:
Merge pull request #3491 from jepler/makefile-translation-exclude

Makefile: Correct and expand translation excluded directories

tannewt committed:
Merge pull request #3469 from jepler/noreturn

Add some NORETURN attributes

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (827 of 827 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/es/

bergdahl committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (827 of 827 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/sv/

tannewt committed:
Merge pull request #3485 from jepler/framebuffer-zero-size

Framebuffer zero size

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/

tannewt committed:
Merge pull request #3500 from weblate/weblate-circuitpython-main

Translations update from Weblate

wtuemura committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (829 of 829 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/pt_BR/

UnexpectedMaker committed:
Removed hardware APA settings from mpconfigboard.h because they make the APA flicker on boot. I need to look into why,  but I don't think they are being used for anything right now anyway.

deshipu committed:
Bump circuitpython-stage to 1.1.1

This adds handling of GameOver exception as SystemExit in anticipation
of the new way of starting games from the menu.

deshipu committed:
Bump pew-pewpew-standalone-10.x to 2.0.1

bergdahl committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (829 of 829 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/sv/

jepler committed:
displayio: Fix matrixportal crash

An RGBMatrix has no bus and no bus_free method.  It is always possible
to refresh the display.

This was not a problem before, but the fix I suggested (#3449) added
a call to core_bus_free when a FramebufferDisplay was being refreshed.
This was not caught during testing.

This is a band-aid fix and it brings to light a second problem in which
a SharpDisplay + FrameBuffer will not have a 'bus' object, and yet does
operate using a shared SPI bus.  This kind of display will need a
"bus-free" like function to be added, or it can have problems like
#3309.

tannewt committed:
Merge pull request #3503 from pewpew-game/main

Bump circuitpython-stage to 1.1.1

ladyada committed:
Merge pull request #3507 from jepler/issue3506-matrixportal-crash

displayio: Fix matrixportal crash

jepler committed:
Update protomatter

This brings in the following, and updates us to the 1.0.4 release tag:

Submodule lib/protomatter 2a1ba8fa4..5f07ec618:
  > Bumping version for release
  > Merge pull request #21 from makermelissa/master
  > Merge pull request #20 from makermelissa/master
  > Merge pull request #18 from jepler/fix-cpy-3184
  > Merge pull request #14 from hierophect/cpy-timer-allocator

We previously had the _changes_ of jepler/fix-cpy-3184 and
hierophect/cpy-timer-allocator but not their merge commits.

The only other changes in protomatter were one formatting change in the
core, plus several Arduino sketches.  So this should make no practical
difference for CPy.

stinos commented:
You should probably create an issue, not a pull request, and definitely not one with 8k commits :)

tve commented:
Surreptitious update of micropython to circuitpython - ROTFL!

nijel committed:
Translated using Weblate (Korean)

Currently translated at 9.4% (78 of 829 strings)

Translation: CircuitPython/main
Translate-URL: https://hosted.weblate.org/projects/circuitpython/main/ko/

jepler committed:
Merge pull request #3509 from weblate/weblate-circuitpython-main

Translations update from Weblate

tannewt committed:
Merge pull request #3510 from jepler/update-protomatter

Update protomatter to 1.0.4 release tag

hierophect committed:
Update ESP-IDF to newer fixed fork

tannewt committed:
Merge pull request #3512 from UnexpectedCircuitPython/UM_S2_Boards

Removed hardware APA settings from ESP32-S2 Feather mpconfigboard.h

tannewt committed:
Merge pull request #3492 from hierophect/esp32-update-idf

ESP32S2: move to official IDF submodule
